src/thread/id.rs

```diff
@@ -40,6 +40,30 @@ pub fn set_thread_uid(uid: Uid) -> io::Result<()> {
     backend::thread::syscalls::setuid_thread(uid)
 }
 
+/// `setresuid(ruid, euid, suid)`
+///
+/// # Warning
+///
+/// This is not the setresxid you are looking for... POSIX requires xids to be
+/// process granular, but on Linux they are per-thread. Thus, this call only
+/// changes the xid for the current *thread*, not the entire process even
+/// though that is in violation of the POSIX standard.
+///
+/// For details on this distinction, see the C library vs. kernel differences
+/// in the [setuid man page][linux_notes]. This call implements the kernel behavior.
+///
+/// # References
+///  - [POSIX]
+///  - [Linux]
+///
+/// [POSIX]: https://pubs.opengroup.org/onlinepubs/9699919799/functions/setuid.html
```

Comment:
It appears `setresuid` and `setresgid` are not in POSIX at all, so we should remove the POSIX references.

Reply:
The POSIX reference was to explain the whole-process requirement, which is why I changed the reference to say "the setuid manpage". Do you want me to remove that? Or perhaps just cross-reference the note under setuid?

Reply:
Just cross-referencing the note under `setuid` sounds good to me.